rustls/src/vecbuf.rs

```diff
@@ -44,11 +44,10 @@ impl ChunkVecBuffer {
 
     /// How many bytes we're storing
     pub(crate) fn len(&self) -> usize {
-        let mut len = 0;
-        for ch in &self.chunks {
-            len += ch.len();
-        }
-        len
+        self.chunks
+            .iter()
+            .map(|ch| ch.len())
```

Comment:
I think this could be `map(Vec::len)`.

Alternatively, maybe the whole `map/sum` should be a fold?

```rust
 self.chunks
            .iter()
            .fold(0usize, |acc, chunk| acc + chunk.len())
            - self.prefix_used
```

I don't feel strongly in either case.